tests/cli-misc.rs

```diff
@@ -129,55 +129,55 @@ fn show_works_with_dumb_term() {
     });
 }
 
-// Issue #140
-// Don't panic when `target`, `update` etc. are called without subcommands.
+// Issue #2427
+// Exit gracefully when called without subcommand.
 #[test]
 fn subcommand_required_for_target() {
     setup(&|config| {
         let mut cmd = clitools::cmd(config, "rustup", &["target"]);
         clitools::env(config, &mut cmd);
         let out = cmd.output().unwrap();
-        assert!(!out.status.success());
-        assert_ne!(out.status.code().unwrap(), 101);
+        assert!(out.status.success());
```

Comment:
so this is saying that this is no longer an error from a scripting point of view. I'm not sure that that makes sense: it has after all not actually accomplished anything, and so for a scripter, having a non-zero exit status is useful, because this presumably happens due to forgetting to put some calculated $thing together properly.

Reply:
I think this is logical, and some tools definitely return an exit code when no subcommands are provided, but Rust tooling doesn't seem to. For example, running `rustc` or `cargo` without subcommands prints the help output and returns a zero `0` exit code.

But I think for the reasons @kinnison shared [here](https://github.com/rust-lang/rustup/pull/2427#issuecomment-662286661), being more consistent with previous versions, is a strong reason to keep the non-zero exit code.

I'll update the PR to keep the exit code.